sql/slave.cc

```diff
@@ -3692,6 +3694,11 @@ has_temporary_error(THD *thd)
   if (!likely(thd->is_error()))
     DBUG_RETURN(0);
 
+  if (rgi->rli->mi->using_parallel() &&
+      rgi->parallel_entry->stop_abrupt(rli) &&
+      rgi->parallel_entry->rgi_is_safe_to_terminate(rgi))
+    DBUG_RETURN(0);
```

Comment:
This logic doesn't belong in has_temporary_error(), it is about whether to stop the slave, not about classification of an error.